ddtrace/tracer/metrics_test.go

```diff
@@ -28,7 +28,9 @@ func TestReportMetrics(t *testing.T) {
 	}
 
 	var tg testGauger
-	go trc.reportMetrics(&tg, time.Millisecond)
+	tg.statsTags = statsTags(trc.config)
```

Comment:
I don't really like this. It seems that it just tests the `statsTags` function from `option.go`. I think this and the associated asserts should be removed, and maybe a dedicated test can be added for `statsTags`.